### PR TITLE
DCGEO-86: Initial implementation of Claim/ClaimResponse integration for pre adjudicated data

### PR DESCRIPTION
**Pull request by KyRobbins**
### Fixes [DCGEO-86](https://jira.cms.gov/browse/DCGEO-86)

Needing to integrate with the BFD service for the new Claim/ClaimResponse resources related to pre adjudicated claims data.

### Proposed Changes

Add new resource types for the bcda worker to process
  - New switch cases for Claim/ClaimResponse resources
    - Alter anonymous function to use models.CCLFBeneficiary to allow access to both MBI and BBID for client invocations.
  - New client methods for retrieving Claim/ClaimResponse resources from BFD
  - Update mock client for test files

### Change Details

Changes set up the framework for a basic call out to get Claim/ClaimResponse data

### Security Implications

Client requests should be made using hashed MBI values, with bundled responses similar to existing client calls.

- [ ] new software dependencies

- [ ] security controls or supporting software altered

- [ ] new data stored or transmitted

- [ ] security checklist is completed for this change

- [ ] requires more information or team discussion to evaluate security implications

- [ ] no PHI/PII is affected by this change

### Feedback Requested

Do the proposed changes follow existing conventions and application standards?
Do the proposed changes seem to fulfill the original request as stated?

### CODE REVIEW

File: bcda/client/bluebutton.go
Comment (ericbuckley) [inline bcda/client/bluebutton.go:181]: let's pass in the ClaimsWindow to these functions as well and set the service-date based on it.  I know the current search functions in BFD will ignore it, but if we do it now it will eventually work once we complete DCGEO-117

Comment (ericbuckley) [inline bcda/client/bluebutton.go:181]: Ditto for the "IncludeTaxNumbers" and "excludeSAMHSA" header values, those will be completed when we implement DCGEO-130 on the BFD

File: bcdaworker/worker/worker.go
Comment (ericbuckley) [inline bcdaworker/worker/worker.go:210]: nice work with the refactor

**Review by ericbuckley — DISMISSED** [review]
looks great

**Review by brianburton — DISMISSED** [review]
LGTM

File: bcda/client/bluebutton.go
Comment (CosmicSyntax) [inline bcda/client/bluebutton.go:205]: I believe the mbi is already hashed. As seen here:
https://github.com/CMSgov/bcda-app/blob/ee32ebfecbc39aa5057cade34b0b9e22854bf646/bcdaworker/worker/worker.go#L197

File: bcda/client/bluebutton.go
Comment (CosmicSyntax) [inline bcda/client/bluebutton.go:185]: Mbi may be already hashed. See below.

File: bcda/client/bluebutton_test.go
Comment (CosmicSyntax) [inline bcda/client/bluebutton_test.go:274]: There are some redundancies in this test suite as a whole, but (while not super necessary) it may be worth adding to the table driven test below. See line 322.

File: bcda/client/bluebutton.go
Comment (ericbuckley) [reply bcda/client/bluebutton.go:205]: I thought the `CCLBeneficiary.MBI` was the raw value, as it appears to be getting hashed here as well https://github.com/CMSgov/bcda-app/blob/ee32ebfecbc39aa5057cade34b0b9e22854bf646/bcdaworker/worker/bluebutton.go#L15

Comment (CosmicSyntax) [reply bcda/client/bluebutton.go:205]: My bad. I see that now as seen here:
https://github.com/CMSgov/bcda-app/blob/fadeccdc727eb55cfc0d3b4cb0946ec8ddcc6aa6/bcdaworker/worker/worker.go#L169
CCLFBeneficiary.MBI is used. Yes that is the raw, and it is perfectly fine.
But we do have CCLFBeneficiary.BlueButtonID which is hashed. Could we use that? Currently, we hashing twice separately.

Comment (ericbuckley) [reply bcda/client/bluebutton.go:205]: I thought the `CCLFBeneficiary.BlueButtonID` was the referencing the CCW BeneId and not the MBI.  Our records don't have a BeneID and thus we need to search directly on the MBI.

Comment (CosmicSyntax) [reply bcda/client/bluebutton.go:205]: Interesting. Apologies - I did not know that. Then this is 🎉 .

File: bcda/client/bluebutton_test.go
Comment (SMLuthi) [reply bcda/client/bluebutton_test.go:274]: Agreed, the table driven tests below are a bit more thorough and it would be beneficial to add Claim and ClaimResponse tests there

**Review by SMLuthi — CHANGES_REQUESTED** [review]
This looks good to me! I just doubled on Daniels comments to add new BB tests to the table!

**Review by ericbuckley — APPROVED** [review]

**Review by SMLuthi — APPROVED** [review]
LGTM! Nice job!

**Comment by Gabasol** [comment]
Hey @CosmicSyntax and @SMLuthi - how did you want to handle DC Geo PR merges?  Did your team want to do the merging, or can our team merge them?  We merge our PRs in the BFD repo...

**Comment by SMLuthi** [comment]
@Gabasol We merge our own PR's once we feel comfortable with the approvals from the reviewers. But I'll defer to @CosmicSyntax on how he wants DC Geo to handle their PR's!

**Review by CosmicSyntax — APPROVED** [review]
🎉 LGTM!

**Pull request merged**
